renderite/serialization.zig

```diff
@@ -155,6 +151,30 @@ pub const IpcDeserializer = struct {
 
         return try self.reader.readSliceEndianAlloc(gpa, T, @intCast(len), endian);
     }
+
+    pub fn readNestedList(self: IpcDeserializer, comptime T: type) !T {
+        const len = try self.reader.takeInt(i32, endian);
+        if (len == 0)
+            return &.{};
+
+        const ChildType = std.meta.Child(T);
+
+        var list = try self.gpa.alloc(ChildType, @intCast(len));
+
+        for (0..list.len) |i| {
+            list[i] = try self.readValueList(std.meta.Child(ChildType), self.gpa);
+        }
```

Comment:
Lots of possible memory leaks here

```suggestion
        var list = try self.gpa.alloc(ChildType, @intCast(len));
        var written: usize = 0;
        errdefer {
        	for(list[0..written]) |sub_list| {
        		self.gpa.free(sub_list);
      	 	} 
        	self.gpa.free(list);
        };
        
        for (list) |*sub_list| {
            sub_list.* = try self.readValueList(std.meta.Child(ChildType), self.gpa);
            written += 1;
        }
```

Reply:
Oof. Glad I made this a dedicated method rather than reworking the existing list methods